solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,501 @@
+########################################################################################################################
+# Input variables
+########################################################################################################################
+
+variable "ibmcloud_api_key" {
+  type        = string
+  description = "The IBM Cloud API key used to provision resources."
+  sensitive   = true
+}
+
+variable "existing_resource_group_name" {
+  type        = string
+  description = "The name of an existing resource group to provision resources in."
+  default     = "Default"
+  nullable    = false
+}
+
+variable "prefix" {
+  type        = string
+  description = "The prefix to add to all resources that this solution creates (e.g `prod`, `test`, `dev`). To not use any prefix value, you can set this value to `null` or an empty string."
+  nullable    = true
+  validation {
+    condition = (var.prefix == null ? true :
+      alltrue([
+        can(regex("^[a-z]{0,1}[-a-z0-9]{0,14}[a-z0-9]{0,1}$", var.prefix)),
+        length(regexall("^.*--.*", var.prefix)) == 0
+      ])
+    )
+    error_message = "Prefix must begin with a lowercase letter, contain only lowercase letters, numbers, and - characters. Prefixes must end with a lowercase letter or number and be 16 or fewer characters."
+  }
+}
+
+variable "provider_visibility" {
+  description = "Set the visibility value for the IBM terraform provider. Supported values are `public`, `private`, `public-and-private`. [Learn more](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/guides/custom-service-endpoints)."
+  type        = string
+  default     = "private"
+  nullable    = false
+
+  validation {
+    condition     = contains(["public", "private", "public-and-private"], var.provider_visibility)
+    error_message = "Invalid value for 'provider_visibility'. Allowed values are 'public', 'private', or 'public-and-private'."
+  }
+}
+
+variable "vsi_resource_tags" {
+  description = "The list of tags to add to the Virtual server instance."
+  type        = list(string)
+  default     = []
+}
+
+variable "vsi_access_tags" {
+  type        = list(string)
+  description = "The list of access tags to add to the Virtual server instance. For more information, see https://cloud.ibm.com/docs/account?topic=account-access-tags-tutorial."
+  default     = []
+}
+
+##############################################################################
+
+
+##############################################################################
+# VPC Variables
+##############################################################################
+
+variable "existing_vpc_crn" {
+  description = "The CRN of an existing VPC. If the user provides only the `existing_vpc_crn` the VSI will be provisioned in the first subnet of the VPC."
+  type        = string
+  nullable    = false
+}
+
+variable "existing_subnet_id" {
+  description = "The ID of an existing subnet."
```

Comment:
The description should specify what happens if no value is passed, since it has a default value of null